lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -224,6 +231,34 @@ def rescuable_exceptions
       def rescue_from
         []
       end
+
+      def extended_cloned_config(tenant)
+        @config.clone.with_indifferent_access.tap do |config|
+          config.merge!(Apartment.db_config_for(tenant))
```

Comment:
Yes thats right, its always merged upon the unchanged config from database.yml :)